### PR TITLE
Player screen

### PR DESCRIPTION
**Pull request by Razor0744**

### CODE REVIEW

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    var isVisible by remember { mutableStateOf(false) }
+
+    val exoPlayer = remember {
+        ExoPlayer.Builder(context).build().apply {
+            this.playWhenReady = true
+        }
+    }
+
+    LaunchedEffect(url) {
+        exoPlayer.setMediaItem(MediaItem.fromUri(url))
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt]: exoPlayer.apply {
...
}

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                override fun onEvents(player: Player, events: Player.Events) {
+                    super.onEvents(player, events)
+                    totalDuration = player.duration.coerceAtLeast(0L)
+                }
+            }
+
+        exoPlayer.addListener(listener)
+
+        onDispose {
+            exoPlayer.removeListener(listener)
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt]: Please, use apply

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/PlayerScreen.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+fun PlayerScreen(
+    modifier: Modifier = Modifier,
+    id: Int,
+    episode: Int,
+    viewModel: PlayerViewModel = koinViewModel()
+) {
+    val uiState by viewModel.uiState.collectAsState()
+
+    LaunchedEffect(Unit) {
+        viewModel.getTitle(id = id)
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/player/presentation/ui/PlayerScreen.kt:22]: Are you sure, that you should fetch title from network? I think that you can pass it from DetailsScreen?

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt (hunk 1 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt]: did

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt (hunk 2 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt]: did

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/PlayerScreen.kt (hunk 3 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/player/presentation/ui/PlayerScreen.kt:22]: yes

File: app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import android.content.pm.ActivityInfo
+
+fun Context.findActivity(): Activity? = when (this) {
+    is Activity -> this
+    is ContextWrapper -> baseContext.findActivity()
+    else -> null
+}
+
+fun Context.setLandscape() {
+    val activity = this.findActivity()
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt]: can be refactored used apply or with

File: app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+}
+
+@SuppressLint("SourceLockedOrientationActivity")
+fun Context.setPortrait() {
+    val activity = this.findActivity()
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt]: can be refactored used apply or with

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt [hunk 6]
```diff
@@ -103,11 +101,15 @@ fun Player(
                 }
             }
 
-        exoPlayer.addListener(listener)
+        exoPlayer.apply {
+            addListener(listener)
```
Comment (MaximSemashko) [inline app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt:121]: here apply is unnesseccary

File: app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt (hunk 4 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt]: did

File: app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt (hunk 5 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/common/extensions/ContextExtension.kt]: did

File: app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt (hunk 6 above)
Comment (Razor0744) [reply app/src/main/java/com/example/movopfy/features/player/presentation/ui/Player.kt:121]: ok

**Pull request merged**